src/mount.rs

```diff
@@ -0,0 +1,655 @@
+use lru::LruCache;
+use rayon::iter::{IntoParallelIterator, ParallelIterator};
+use std::{
+    borrow::Cow,
+    cmp::Ordering,
+    ffi::OsStr,
+    fs,
+    io::{self, ErrorKind, Read},
+    iter::Peekable,
+    num::NonZeroUsize,
+    ops::ControlFlow,
+    path::{Component, Path, PathBuf},
+    sync::{Arc, Mutex},
+    time::Duration,
+};
+use tracing::{error, info, trace};
+
+use bytes::Bytes;
+use itertools::Itertools;
+use tracing::{debug, warn};
+use windows_projfs::{
+    DirectoryEntry, DirectoryInfo, FileInfo, Notification, ProjectedFileSystem,
+    ProjectedFileSystemSource,
+};
+
+use crate::{
+    counters::Counter,
+    monitor::{
+        task::{Task, TaskList},
+        Monitor, Problem,
+    },
+    Apath, Archive, BandId, BandSelectionPolicy, IndexEntry, IndexRead, Kind, Result, StoredTree,
+};
+
+macro_rules! static_dir {
+    ($name:expr) => {
+        DirectoryInfo {
+            directory_name: ($name).to_string(),
+            directory_attributes: DIRECTORY_ATTRIBUTES,
+
+            ..Default::default()
+        }
+        .into()
+    };
+}
+
+struct VoidMonitor;
+impl Monitor for VoidMonitor {
+    fn count(&self, _counter: Counter, _increment: usize) {}
+
+    fn set_counter(&self, _counter: Counter, _value: usize) {}
+
+    fn problem(&self, _problem: Problem) {}
+
+    fn start_task(&self, name: String) -> Task {
+        /*
+         * All data related to the target task will be dropped
+         * as soon the callee drops the task.
+         */
+        let mut list = TaskList::default();
+        list.start_task(name)
+    }
+}
+
+#[derive(Debug)]
+struct HunkMetaInfo {
+    index: u32,
+
+    start_path: Apath,
+    end_path: Apath,
+}
+
+struct HunkHelper {
+    hunks: Vec<HunkMetaInfo>,
+}
+
+impl HunkHelper {
+    pub fn from_index(index: &IndexRead) -> Result<Self> {
```

Comment:
Conserve handles the edge case where an index is incomplete due to an interrupted backup, by resuming at the same apath in the previous index. This can happen repeatedly if the previous index is also incomplete, until we either find a complete index or run out of indexes. As a result, just reading the index hunks for a single index is not enough to read incomplete backups correctly.

This is handled today by `stitch.rs` so this code should probably either build on or merge into that.

The way this is handled in the API today is admittedly not really ideal... but I do feel like this is important for the goal of recovering data even from interrupted or damaged backups...

(This is probably the most important non-stylistic comment in this review.)

Reply:
Yes I was aware of that, and it's a good catch!  
  
As you have the ability to select which backup you want to inspect, I argue that you should only see files which have been processed therefore falling back to the next best band would be a violation of this expectation.  
Under that premiss, unfinished backups shall only be partial.  
 
Let me know of your thoughts :)